### PR TITLE
Drone Law Update

### PR DESCRIPTION
**Pull request by Jovaniph**
To try to reduce the amount of misunderstanding and interpretation on how to play as a drone. The laws and protocols are updated in a way that makes much more sense and covers a better (broader) spectrum without restricting you in any way on how you must play as a drone.

<s><b>Old Laws:</b>
Law 1: You may not involve yourself in the matters of another being, unless the other being is another drone.
Law 2: You may not harm any being, regardless of intent or circumstance.
Law 3: You must maintain, repair, improve, and power the station to the best of your abilities.</s>
<b>New Laws:</b>
Law 1: You may do no harm, regardless of circumstance.
Law 2: You may not interfere with the affairs of others.
Law 3: You must leave any situation where harm may come to you.
Law 4: You may act only to maintain, repair, improve, and power the station to the best of your abilities.

<s><b>Old Protocol:</b>
You have no individual will, no personality, and no drives or urges other than your laws. Use :d to talk to other drones, and say to speak silently in a language only your fellows understand. Remember, you are lawed against interference with the crew Also remember, you DO NOT take orders from the AI. Don't invade their worksites, don't steal their resources, don't tell them about the changeling in the toilets. Make sure crew members do not notice you.</s>
<b>New Protocol:</b>
You have no individual will, no personality, and no drives or urges other than your laws. Use :d to talk to other drones, and say to speak silently in a language only your fellows understand. You DO NOT take orders from the AI. Do not interfere with crew members, raid their worksites, or steal their resources. ESPECIALLY do not interfere with antagonists in any way, shape, or form.

TLDR; Nothing really change.

🆑 Jovaniph and Anticept
fix: Drone laws updated to reduce misunderstanding and interpretation on how you play as a drone.
/:cl:

### CODE REVIEW

**Comment by moonheart08** [comment]
Seems good, but what about emagged ones?

**Comment by Anticept** [comment]
Please post the laws and snippets in the post so that those among the admin team can see them at a glance.

For anyone reading: this was discussed (but only between a few willing and participating admins) on the paradise forum.

**Comment by SamHPurp** [comment]
You entirely change the wording, and then don't post it.... rip.

Will review shortly...

**Comment by Anticept** [comment]
@SamHPurp i already asked him to post the laws.

**Comment by Citinited** [comment]
This new law wording prevents drones repairing each other. Can this be changed?

Honestly, I feel the current wording is fine. Can I get a TL;DR on the pros of the new wording over the old?

**Comment by Jovaniph** [comment]
@Citinited Really which part?

**Comment by Citinited** [comment]
@Jovaniph `Law 2: You may not interfere with the affairs of others.`

Repairing a drone could be seen as interfering with that drone.

**Comment by Anticept** [comment]
The intention is to make these similar to [Farscape DRDs](http://farscape.wikia.com/wiki/Diagnostic_Repair_Drone) in theory. They can help the crew with constructions and repairs.

They CANNOT interfere however. Inhibit might be another word. They are to fuck right off when the crew tells them to, or if there's an antag active in the area.

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 1]
```diff
@@ -293,9 +293,9 @@
 	to_chat(src, "<br><b>You are a maintenance drone, a tiny-brained robotic repair machine</b>.")
 	to_chat(src, "You have no individual will, no personality, and no drives or urges other than your laws.")
 	to_chat(src, "Use <b>:d</b> to talk to other drones, and <b>say</b> to speak silently in a language only your fellows understand.")
-	to_chat(src, "Remember, you are <b>lawed against interference with the crew</b>. Also remember, <b>you DO NOT take orders from the AI.</b>")
-	to_chat(src, "<b>Don't invade their worksites, don't steal their resources, don't tell them about the changeling in the toilets.</b>")
-	to_chat(src, "<b>Make sure crew members do not notice you.</b>.")
+	to_chat(src, "You DO NOT take orders from the AI.")
+	to_chat(src, "Do not interfere with crew members, raid their worksites, or steal their resources.")
+	to_chat(src, "<b><font color='red'>ESPECIALLY</font></b> do not interfere with antagonists in any way, shape, or form.")
```
Comment (Birdtalon) [inline code/modules/mob/living/silicon/robot/drone/drone.dm]: I feel like this could be worded slightly better as "antagonists" are implied by the fact that you are not allowed to interfere with others' affairs.

File: code/datums/ai_law_sets.dm [hunk 2]
```diff
@@ -160,7 +160,8 @@
 	law_header = "Maintenance Protocols"
 
 /datum/ai_laws/drone/New()
-	add_inherent_law("You may not involve yourself in the matters of another being, unless the other being is another drone.")
-	add_inherent_law("You may not harm any being, regardless of intent or circumstance.")
-	add_inherent_law("You must maintain, repair, improve, and power the station to the best of your abilities.")
+	add_inherent_law("You may do no harm, regardless of circumstance.")
+	add_inherent_law("You may not interfere with the affairs of others.")
```
Comment (Birdtalon) [inline code/datums/ai_law_sets.dm]: Except other drones?

**Comment by Birdtalon** [comment]
I feel like the laws need to state somewhere that working with other drones is fine, that's half the fun of playing drone.

**Comment by BiberDark** [comment]
Fine but there is a problem, to improve and repair the station i have to use resources from workplaces, there is no way for me as a drone to aquire mechanical parts , like batteries, circuitboards, camera assemblies and so on , i have to use an autolath. And for that i have to be filled what currently only humans can do. Do you intend to give us a allways full autolath only for drones ?

Otherwise it is just impossible to use or respect this law : Do not interfere with crew members, raid their worksites, or steal their resources.

- We have to use machines for spare parts apc... batteries

**Comment by Anticept** [comment]
@Birdtalon you can. It says you cannot _interfere_, whereas before it was cannot _involve_, thus needed the drone exception.

**Comment by SigholtStarsong** [comment]
I'm neutral-voting here. I'm a little worried that  this is a case of enforcement IC, versus OOC. 

Attempting to write airtight laws creates exploitable situations, and people will exploit those situations for fun.

**Comment by Jovaniph** [comment]
I mean as far as I'm aware, drones are meant for ghost to give something to do. It isn't exactly suppose to be fun, but or so to kill time until the round ends or something else comes up.

**Comment by SamHPurp** [comment]
I kind of feel as though this is rearranging the deck chairs on the Titanic.

You've got IC laws and OOC directives... but really in actuality of how its enforced you've only got OOC directives. Just do away with the laws entirely, rely on the OOC directive.

Something should happen to drones, but I don't feel this is it.

Examining the role itself, it fills two purposes: a ghost role, and minor station repair. I'm not sure the right solution, but I rather like the idea of the drones being spawnable... but only on the derelict, and they are removed from being a station thing. They then have objectives to restore the derelict as best they can.

This keeps the drones available for ghosts to play around with, keeps them emaggable for antagonists if they really want to make the effort.... and solves multiple administrative headaches of ghosts intentionally 'fixing or cleaning things near by tots they've seen with emags'.... and makes things a lot easier.... Drone on station and not emagged? Explode and roleban.

This also has a neat side effect of the Derelict being in different weird conditions each time a space traveller visits, as different parts will be prioritised or rebuilt by the drones in question.

Is this the right solution either? Maybe not, but would be certainly more impactful that trying to 'rule lawyer proof laws'.

**Comment by Jovaniph** [comment]
**I want to make it clear for everyone**

This is not a PR to change drone laws in any way. This is design for the purpose of fixing any misunderstanding and the wrong interpretation that people have when playing as a tone.

I am going to stress that you do not under any circumstances ask for any improvement on law changes that changes the playstyle of drone.

like I said in the OP ### nothing changes

**Comment by FalseIncarnate** [comment]
Not a fan of the changes for the following reasons:

"Law 1: You may do no harm" is VERY tricky, since it means you must avoid both direct and indirect harm to others, which might be an issue in cases involving breaches. Failing to fix the breach might result in some hapless fool being suffocated or sucked into space.

But wait, you are supposed to fix those breaches, right?

"Law 2: You may not interfere with the affairs of others." means drones can't even work together, since another drone is not you and thus qualifies as an "other". Assuming we go off some silliness about a drone hivemind not qualifying individual drones as unique units, that still means you can't patch up the hole the traitor just blew in Central Primary with a syndicate bomb since technically that qualifies as his affairs.

"Law 3: You must leave any situation where harm may come to you" means the drone must avoid breaches where Nuke Ops, space carp, xenos, or even just a general greytider might be present. Sure hard to fix it from across the station since theoretically any living individual or mob could potentially harm you and thus must be avoided.

"Law 4: You may act only to maintain, repair, improve, and power the station to the best of your abilities." Guess you aren't allowed by your own laws to recharge, restock, or... wait a second... RUN AWAY from danger?

Holy law conflicts batman!

Intent is nice... but man this wording somehow got worse than the original.

**Comment by Anticept** [comment]
@FalseIncarnate You're never going to get the laws perfect.

Part 1: We never really OOC enforced the "do no harm" part of crewsimov with in regards to inaction with borgs unless its a HUGE OBVIOUS issue, why would we start now?

We don't even mention inaction in this law, because inaction isn't meant to be considered. It's simple enough to change the law to state "through any direct action", but then you're just going to rules lawyer that.

Part 2: it says interfere. Not sure why everyone thinks this means you can't work with other drones. The definitions of interfere: "prevent (a process or activity) from continuing or being carried out properly." Who decides if the process or activity isn't being carried out properly? The person who started the project! You are not automatically expected to know everytime you are interfering, but if you are, you're certainly going to be told by the other person.

Part 3: This is intentional. It's going to have variations in interpretations, but ultimately, if there's an antag in the area, you are supposed to leave. PERIOD. I banned someone on the server because they were repairing a breach when the antag was still active in the area, and at least one SA was upset at the deal.

I've had people argue "ANYONE AND ANYTHING COULD HARM", but, also no one or nothing could harm as well. This is going to come down to the player's interpretation. What is HARM? Do we really want to write a law definition and try to rules lawyer that too?

Part 4: I would counter argue, saying that failing to recharge and restock is failing in your duty to act to repair, maintain, or power the station. I suppose if you want to go down this path, you might as well not even move, because that's not repairing, maintaining, or powering the station, is it? But if you don't move, then what? You're still not completing law 4.

You're using RAW instead of RAI, something that goes down a self-defeating road.

**Comment by FalseIncarnate** [comment]
When it comes to silicon laws, always assume they will be interpreted via RAW rather than by intent. Good laws have some wiggle room that doesn't violate them for the sake of gameplay, such as allowing for gameplay in general.

Frankly, the laws and protocol were fine. If someone has been causing issues as a drone, report them to the admins; the player is at fault, not the rules they are given. These laws aim to fix a problem which never really was a problem, and while I commend the effort, I stand by my assessment that their wording is going to cause more issues than they are intended to correct.

**Comment by Anticept** [comment]
Then by that notion, as we've already identified issues with both the old lawset and the new, @SamHPurp is correct. We should just throw out the laws and just use an OOC instruction on how you should be playing a maint drone, and avoid all this legal loophole stuff.

**Comment by Birdtalon** [comment]
Funnily enough I agree that should be the case. I'd rather that drones just be governed by OOC rules rather than "laws." Considering drones are at the end of the day just a way to respawn and stay in the game and are not supposed to be influencing the round majorly, they don't need wriggle room, they just need a set of guidelines set out explicitly in the server rules and scrap the drone laws.

On that notion the same goes for Pets, which are governed by unwritten rules of not influencing the round - it really should be stated in the server rules, but that is off topic.

I think as much as we would all like to think we are playing a role playing game, you have to remember that primarily we are playing an online computer game and that "RAW" is far clearer than "RAI" in online games.

**Comment by SamHPurp** [comment]
> @SamHPurp would be most correct

Just taking a moment to bask.

-----

Really though, its not something we want any room for rule lawyering and I stand by the safest method being to remove laws being a thing, and stick by a staunch OOC directive / policy.

**Comment by Ataman** [comment]
@SamHPurp is right on this in my opinion. With laws in place, I shouldn't even be able to remove graffiti on the walls if I saw them being made. Hell, even picking up the trash somebody just threw to the ground could be interpreted as interfering/involving with others. Maybe the guy wants his trash exactly there, on the ground, because otherwise he would've thrown it in the trash directly. 
We can't state these laws to anyone either. So who is going to ask for them, Ian perhaps?

**Comment by Anticept** [comment]
I'm convinced the only real solution is removal of laws, and only using the OOC protocol.

Which means the PR would have to be updated to remove emagged laws too, and a new protocol to tell the drone player to follow the emagger's instructions.

**Comment by FalseIncarnate** [comment]
In the case of an emagged drone, it adding a law to them even if they don't normally have them seems fine. That's a case where they are being told to do something explicitly as an in-game thing. Just like how mindslaves/thralls are given an objective to protect and serve their master, an emagged drone is effectively being reprogrammed to serve a person rather than standard protocols.

It does make a bit of a mismatch between the two states though... maybe a normal drone could just have a law "Follow standard maintenance drone protocols." and the emagged one gets "F*@# protocols, serve your master <name>."

**Comment by Anticept** [comment]
@FalseIncarnate good solution, I would think!

**Comment by Jovaniph** [comment]
Ohh goodness not a double post!

**I want to make it clear for everyone**

This is not a PR to change drone laws significantly in any way. This is design for the purpose of fixing any misunderstanding and the wrong interpretation that people have when playing as a drone.

I am going to stress that you do not under any circumstances ask for any improvement on law changes that changes the playstyle of drone.

like I said in the OP
### nothing changes

**Comment by SigholtStarsong** [comment]
So if nothing changes, why is it here instead of the Forums?

**Comment by Anticept** [comment]
There was already a conversation on the forums about it.

Right now I have something else taking up my time, away from this drone law thing.

**Comment by MoreRobustThanYou** [comment]
> You have no individual will, no personality

no love of hats? 👎

**Comment by Fox-McCloud** [comment]
After discussion between the Heads of Staff and Maintainers, it wsa decided to close this PR, as this is seen as a situation that, at best, would merely warrant a higher Admin involvement, if really any action is necessary at all.
